### DISCUSSION TITLE
Adding the OID Pantum

### DISCUSSION THREAD

**Discussion by Krovyakov-Ivan** [glpi-project/glpi-agent#590] [Ideas]
### Your idea

Good afternoon! New Pantum printers are appearing in our network. Is it possible to add the IDs of these devices to the inventory? The counter is very important.

**Comment by g-bougard** [comment]
Hi @Krovyakov-Ivan 

can you provide a [snmpwalk output in the right format](https://glpi-agent.readthedocs.io/en/latest/bug-reporting.html#snmpwalk-output) so we can investigate ?

**Reply by Krovyakov-Ivan** [reply]
[Walk-Pantum.zip](https://github.com/glpi-project/glpi-agent/files/14083634/Walk-Pantum.zip)
Good afternoon.
Thank you for responding.
The file was generated according to the instructions (in the attachment)

**Reply by Krovyakov-Ivan** [reply]
this file may also help 
[Pantum.txt](https://github.com/glpi-project/glpi-agent/files/14083777/Pantum.txt)

**Reply by g-bougard** [reply]
Thank you, I'm checking it.
Do you have the MIB of your printer ? This can help to optimize the development of a dedicated module.
Anyway, can you you tell me what's the serial number of that printer ? This is the minimum to have the printer imported automatically. I see a string which seems promising: `CK3U016552`, but it can be catch on few OIDs, if it's the good string, the MIB could help to identify the best OID to use. I see also some strings that could be related to the firmware or software version. For cartridges and counters, I think the mib is required.

**Reply by Krovyakov-Ivan** [reply]
I wrote to the support of Pantum, I hope they will send me a MIB file. So far, I can only provide screenshots from the web interface of Pantum and GLPI-Inventory.
[Pantum.zip](https://github.com/glpi-project/glpi-agent/files/14093785/Pantum.zip)

**Reply by g-bougard** [reply]
Thank you, I still can try to catch serialnumber. I'll also check for page counters, but I'll make assumptions that could be wrong for other models.

**Reply by g-bougard** [reply]
Actually I can only safely catch the serialnumber. You can download the new module from here: [Pantum.pm](https://github.com/glpi-project/glpi-agent/raw/2b24f17984a58502ee8b26121f520fc89974c452/lib/GLPI/Agent/SNMP/MibSupport/Pantum.pm) and put it under the `lib/GLPI/Agent/SNMP/MibSupport` in your glpi-agent installation. Or you can wait for the next nightly build.

**Reply by Krovyakov-Ivan** [reply]
Yes, the serial number has appeared (screenshot in the archive. For some reason I can't add just jpg)
[GLPI1Pantum.zip](https://github.com/glpi-project/glpi-agent/files/14097652/GLPI1Pantum.zip)

**Reply by Krovyakov-Ivan** [reply]
Good afternoon. Support has not yet responded to me about the MIB. I used the sample template for Zabbix(https://github.com/zbx-sadman/zabbix-templates/blob/main/Printers/Pantum/template_pantum_mono_laser_multifunction_printer_snmp/6.0/template_pantum_mono_laser_multifunction_printer_snmp.xml), mapped the OID to my Pantum model. I suggest adding the following OIDs:
Pages printed total           1.3.6.1.4.1.40093.1.1.3.12
Catrirdge resource (%)     1.3.6.1.4.1.40093.6.3.1
Memory size		         1.3.6.1.4.1.40093.1.1.1.2

**Reply by g-bougard** [reply]
Hi,
will check it, thank you for this find.

**Reply by g-bougard** [reply]
Hi @Krovyakov-Ivan 

here is an [update of Pantum.pm](https://github.com/glpi-project/glpi-agent/raw/35166a27de50ff61e5a36ef374dd4782ed7037c1/lib/GLPI/Agent/SNMP/MibSupport/Pantum.pm)

It adds blacktoner , memory, firmware and pagecounters support. For pagecounters, I made few assumption and you'll have to tell me if I'm right after you check the results for your Pantum printers. So I used these OIDs:
* `.1.3.6.1.4.1.40093.1.1.3.1` for 'DUPLEX'
* `.1.3.6.1.4.1.40093.1.1.3.9` for 'COPYTOTAL'
* `.1.3.6.1.4.1.40093.1.1.3.12` for  'PRINTTOTAL'
* I'm making the total of those 3 OID values to set 'TOTAL'

I made this choice for TOTAL as I see .1.3.6.1.4.1.40093.1.1.3.11 is the total for the cartridge. For the provided walk, that total is 15. Zabbix template tells .1.3.6.1.4.1.40093.1.1.3.1 is for duplex and .1.3.6.1.4.1.40093.1.1.3.12 is for total print pages. For these 2 OIDs, I see 1 and 11 for a total of 12. As I see .1.3.6.1.4.1.40093.1.1.3.9 has 3 as value and your printer can make copy I just assume this is the total of copies. This is the assumption I made to match the total of 15 for the cartridge print total. I think I can't use .1.3.6.1.4.1.40093.1.1.3.11 for TOTAL as I understand it is probably reset to 0 when the cartridge is changed... or maybe this is the real total I should just take. But I guess this is a brand new printer regarding the page count and the toner level at 100%. So I still can't assume that as Zabbix seems to tell something else even if they used reverse engeneering. Here the MIB would have been the right source of information.
If I'm wrong, we can remove COPYTOTAL & TOTAL and just keep DUPLEX & PRINTTOTAL.

**Reply by Krovyakov-Ivan** [reply]
Okay, I'll check it out tomorrow. 
And today I discovered that the page counter is incorrectly specified for one of the HP printer models. I think I'll create a separate topic later to address this issue.

**Reply by Krovyakov-Ivan** [reply]
Hi. I replaced the new file that you sent. The archive contains the results that I received. At the beginning, I took the current readings from the printer. Then I printed out a two-sided document (duplex). And I collected the meter readings again.
[Duplex print.zip](https://github.com/glpi-project/glpi-agent/files/14136370/Duplex.print.zip)

**Reply by g-bougard** [reply]
Hi @Krovyakov-Ivan 
thank you for this sharing. I can now assume duplex counter is not behind .1.3.6.1.4.1.40093.1.1.3.1 but more probably behind .1.3.6.1.4.1.40093.1.1.3.20.
I think also .1.3.6.1.4.1.40093.1.1.3.10 & .1.3.6.1.4.1.40093.1.1.3.11 are the counters for toner & drumkit. I think drumkit may also be inventoried, have to dig a little.
Anyway, the total seems wrong too as your screenshots show it goes from 43 to 45. My assumption regarding its computation is surely wrong.
Actually, I'm only confident PRINTTOTAL is right for page counters.
Can you provide also the screenshots for scan & copy from the printer UI ?

**Reply by Krovyakov-Ivan** [reply]
I took screenshots. snmpwalk did the same after each change. Signed the folders.
[Scan and Copy.zip](https://github.com/glpi-project/glpi-agent/files/14138950/Scan.and.Copy.zip)

**Reply by Krovyakov-Ivan** [reply]
Hi. I haven't heard from you for a long time

**Reply by g-bougard** [reply]
Hi @Krovyakov-Ivan 

sorry but I was busy with a lot of other important stuffs. I still have to investigate a little more your last datas before updating the Pantum module.

Any answer about your MIB files request to manufacturer support ?

**Reply by Krovyakov-Ivan** [reply]
I never received an answer, I wrote several letters

**Reply by Krovyakov-Ivan** [reply]
Good afternoon. Support has sent a file
[Pantum MIB data- BP5100 BM5100 series.xlsx](https://github.com/glpi-project/glpi-agent/files/14294235/Pantum.MIB.data-.BP5100.BM5100.series.xlsx)

**Reply by Krovyakov-Ivan** [reply]
Good afternoon. Do you have time to return to my question?

**Reply by Krovyakov-Ivan** [reply]
> Hi @Krovyakov-Ivan 
> 
> sorry but I was busy with a lot of other important stuffs. I still have to investigate a little more your last datas before updating the Pantum module.
> 
> Any answer about your MIB files request to manufacturer support ?

Hi. I have not received an answer from you. So will you help?

**Reply by Krovyakov-Ivan** [reply]
Hi @g-bougard 
Tell me, did you manage to add a MIB for Pantum?
I don't know why you stopped answering me.

**Reply by g-bougard** [reply]
Hi @Krovyakov-Ivan 
sorry, I had other important subjects like the recently security issue fixes.
I'm not sure to have time to come back on that subject before next release, but I'll try.

**Reply by Krovyakov-Ivan** [reply]
Hi @g-bougard 
Please return to my question as soon as possible. The addition of the company Pantum is relevant not only for me.

**Reply by sti-mario** [reply]
In my answer, I have monitored what I did by doing a complete walk analysis to be able to follow these printer indicators. The model is the Pantum BM5100FDW. It has returned the accurate values ​​so far.

**Comment by alex2beard** [comment]
> Good afternoon. Support has sent a file [Pantum MIB data- BP5100 BM5100 series.xlsx](https://github.com/glpi-project/glpi-agent/files/14294235/Pantum.MIB.data-.BP5100.BM5100.series.xlsx)

That's great! I have checked all these OIDs on the Pantum M7310DN, they are all true! I'm really looking forward to entering this data into the agent :)

**Comment by sti-mario** [comment]
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/DANIEL/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/DANIEL/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/DANIEL/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=PT-BR style='tab-interval:35.4pt;word-wrap:break-word'>


This is what I managed to test with the Pantum BM5100FDW Series printer
OIDSNMP | Function
-- | --
1.3.6.1.4.1.40093.1.1.3.10 | Total number of pages printed by the current toner
1.3.6.1.4.1.40093.10.3.1.1 | Global counter of pages printed by the printer
1.3.6.1.4.1.40093.6.3.1 | Remaining toner level (%)
1.3.6.1.4.1.40093.10.1.5.2.1.4.4 | Theoretical toner yield (maximum estimated pages)
1.3.6.1.4.1.40093.10.1.5.2.1.5.4 | Installed Toner Model
1.3.6.1.4.1.40093.8.1.4 | Remaining Drum Level (%)
1.3.6.1.4.1.40093.8.1.3 | Installed Drum Model




</body>

</html>

**Reply by g-bougard** [reply]
Please, use english. This won't help us to help you otherwise.

**Reply by sti-mario** [reply]
OK

**Reply by g-bougard** [reply]
Can you provide a full snmpwalk output of your printer following this doc ? https://glpi-agent.readthedocs.io/en/latest/bug-reporting.html#snmpwalk-output
In your shared OIDs, I can see some differences from what current Pantumo module can handle, but I really need a walk to not break previously supported models.

**Reply by Krovyakov-Ivan** [reply]
It is better to request an official document from Pantum by ID for this model.

**Comment by Krovyakov-Ivan** [comment]
> <html xmlns:o="urn:schemas-microsoft-com:office:office"
> xmlns:w="urn:schemas-microsoft-com:office:word"
> xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
> xmlns="http://www.w3.org/TR/REC-html40">
> 
> <head>
> 
> <meta name=ProgId content=Word.Document>
> <meta name=Generator content="Microsoft Word 15">
> <meta name=Originator content="Microsoft Word 15">
> <link rel=File-List
> href="file:///C:/Users/DANIEL/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
> 
> <link rel=themeData
> href="file:///C:/Users/DANIEL/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
> <link rel=colorSchemeMapping
> href="file:///C:/Users/DANIEL/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">
> 
> <style>
> 
> </style>
> 
> </head>
> 
> <body lang=PT-BR style='tab-interval:35.4pt;word-wrap:break-word'>
> 
> 
> Este é o que consegui fazendo testes com a impressora Pantum BM5100FDW Series
> OID SNMP | Função 
> -- | -- 
> 1.3.6.1.4.1.40093.1.1.3.10 | Número total de páginas impressas pelo toner atual 
> 1.3.6.1.4.1.40093.10.3.1.1 | Contador global de páginas impressas pela impressora 
> 1.3.6.1.4.1.40093.6.3.1 | Nível do toner restante (%) 
> 1.3.6.1.4.1.40093.10.1.5.2.1.4.4 | Capacidade teórica   do toner (número máximo de páginas estimadas) 
> 1.3.6.1.4.1.40093.10.1.5.2.1.5.4 | Modelo do toner   instalado 
> 1.3.6.1.4.1.40093.8.1.4 | Nível do drum restante (%) 
> 1.3.6.1.4.1.40093.8.1.3 | Modelo do drum instalado 
> 
> 
> 
> 
> </body>
> 
> </html>
> 

Boa tarde. Acima está o arquivo que o Support pantum Enviou, com todos os OIDs

**Reply by g-bougard** [reply]
Please, use english. This won't help us to help you otherwise.

**Comment by AshwiDeshbpande** [comment]
Unable to get MIB file for Patum M7100DN Series.please help to share. Need to configure on system
